src/lib/ui/SupportedDevices/SupportedDevicesFilterArea.tsx

```diff
@@ -0,0 +1,163 @@
+import { capitalize } from '@mui/material'
+import type { Dispatch, SetStateAction } from 'react'
+
+import { CloseIcon } from 'lib/icons/Close.js'
+import { SearchIcon } from 'lib/icons/Search.js'
+import { Button } from 'lib/ui/Button.js'
+import Menu from 'lib/ui/Menu/Menu.js'
+import { FilterCategoryMenu } from 'lib/ui/SupportedDevices/FilterCategoryMenu.js'
+import type { DeviceModel, Filters } from 'lib/ui/SupportedDevices/types.js'
+import { TextField } from 'lib/ui/TextField/TextField.js'
+
+interface SupportedDevicesFilterAreaProps {
+  deviceModels: DeviceModel[]
+  filterStr: string
+  setFilterStr: (filterStr: string) => void
+  filters: Filters
+  setFilters: Dispatch<SetStateAction<Filters>>
+}
+
+export function SupportedDevicesFilterArea({
+  deviceModels,
+  filterStr,
+  setFilterStr,
+  filters,
+  setFilters,
+}: SupportedDevicesFilterAreaProps) {
+  const appliedFilters = Object.keys(filters).filter((key) => {
+    return (
+      filters[key as keyof Filters] !== undefined &&
+      filters[key as keyof Filters] !== null &&
+      filters[key as keyof Filters] !== false
+    )
+  })
+  const appliedFiltersCount = appliedFilters.length
+
+  function getAvailablePropertiesFromDeviceModels(property: keyof DeviceModel) {
+    const properties = new Set<string>()
+    deviceModels.forEach((deviceModel) => {
+      properties.add(capitalize(deviceModel[property]))
+    })
+    return Array.from(properties)
+  }
+
+  function resetFilter(filterType: keyof Filters) {
+    setFilters((filters) => ({
+      ...filters,
+      [filterType]: null,
+    }))
+  }
+
+  return (
+    <div className='seam-supported-devices-filter-area'>
+      <div className='seam-deliberate-block' />
+      <div className='seam-filters-wrap'>
+        <Menu
+          button={({ open }) => (
+            <Button
+              variant='outline'
+              className='seam-filters-button'
+              onClick={open}
+            >
+              Filter
+              {appliedFiltersCount !== 0 ? `s (${appliedFiltersCount})` : ''}
```

Comment:
Can we follow this pattern? https://github.com/seamapi/react/commit/3eed1df6f4d6ef04e24fbb84876228401bec5120 

See also https://github.com/seamapi/react/pull/174#discussion_r1204966135